WordPress/WordPressApi/WordPressComRestApi.swift

```diff
@@ -272,6 +247,163 @@ public class WordPressComRestApi: NSObject
+        }
+        fileHandle.writeData("--\(boundary)--\r\n".dataUsingEncoding(NSUTF8StringEncoding)!)
+        fileHandle.closeFile()
+        return request
+    }
+
+    //MARK: - Progress reporting
+
+    private func associate(progress:NSProgress, task: NSURLSessionTask) {
+        // Progress report
+        progress.totalUnitCount = 1
+        if let contentLengthString = task.originalRequest?.allHTTPHeaderFields?["Content-Length"],
+            let contentLength = Int64(contentLengthString) {
+            progress.totalUnitCount = contentLength + 1
+        }
+        progress.cancellationHandler = {
+            task.cancel()
+        }
+        ongoingProgress[task] = progress
+    }
+
+    //MARK: - Response handling
+
+    private func handleResponseWithData(originalData: NSData?, urlResponse:NSURLResponse?, error: NSError?) throws -> AnyObject {
+        guard let data = originalData,
+            let httpResponse = urlResponse as? NSHTTPURLResponse,
+            let contentType = httpResponse.allHeaderFields["Content-Type"] as? String
+            where error == nil &&  contentType.hasPrefix("application/json") else {
+                if let unwrappedError = error {
+                    throw convertError(unwrappedError, data: originalData)
+                } else {
+                    throw convertError(WordPressComRestApiError.ResponseSerializationFailed as NSError, data: originalData)
+                }
+        }
+
+        let jsonObject: AnyObject
+
+        do {
+            jsonObject = try NSJSONSerialization.JSONObjectWithData(data, options: NSJSONReadingOptions())
+        } catch {
+            throw convertError(WordPressComRestApiError.ResponseSerializationFailed as NSError, data: originalData)
+        }
+        if (200..<300).contains(httpResponse.statusCode) {
+            return jsonObject
+        }
+        if (400...500).contains(httpResponse.statusCode) {
+            throw handleErrorResponseObject(jsonObject, originalData: data)
+        }
+        throw convertError(WordPressComRestApiError.ResponseSerializationFailed as NSError, data: originalData)
+    }
+
+    //MARK: - Error Handling
+
+    private func handleErrorResponseObject(responseObject: AnyObject, originalData:NSData) -> NSError {
+        guard let responseDictionary = responseObject as? [String:AnyObject] else {
+            return convertError(WordPressComRestApiError.ResponseSerializationFailed as NSError, data: originalData)
+        }
+        var errorDictionary:AnyObject? = responseDictionary
+        if let errorArray = responseDictionary["errors"] as? [AnyObject] where errorArray.count > 0 {
+            errorDictionary = errorArray.first
+        }
+        guard let errorEntry = errorDictionary as? [String:AnyObject],
+            let errorCode = errorEntry["error"] as? String,
+            let errorDescription = errorEntry["message"] as? String
+            else {
```

Comment:
No ideas if we have a style guide for this, but in this case i normally promote the `{` to a new line, to ease reading